Sources/FaunaDB/Value.swift

```diff
@@ -221,24 +221,81 @@ extension DateV: Equatable {
     }
 }
 
+public class RefID: AsJson {
+    public let id: String
+    public let `class`: RefV?
+    public let database: RefV?
+
+    public init(id: String, class: RefV?, database: RefV?) {
+        self.id = id
+        self.`class` = `class`
+        self.database = database
+    }
+
+    func escape() -> JsonType {
+        var dic: [String: JsonType] = ["id": .string(id)]
+
+        if `class` != nil {
+            dic["class"] = `class`.escape()
+        }
+
+        if database != nil {
+            dic["database"] = database.escape()
+        }
+
+        return .object(dic)
+    }
+}
+
 /// Represents a Ref returned by the server.
 /// [Reference](https://fauna.com/documentation/queries#values-special_types).
 public struct RefV: ScalarValue, AsJson {
 
-    public var value: String
+    public var value: RefID
 
-    public init(_ value: String) {
-        self.value = value
+    public init(_ id: String, class: RefV? = nil, database: RefV? = nil) {
+        self.value = RefID(id: id, class: `class`, database: database)
+    }
+
+    public var description: String {
+        let cls = value.class.map { ", class=\($0)" } ?? ""
+        let db = value.database.map { ", database=\($0)" } ?? ""
+        return "RefV(id=\(value.id)\(cls)\(db))"
     }
 
     func escape() -> JsonType {
-        return .object(["@ref": .string(value)])
+        return .object(["@ref": value.escape()])
     }
 }
 
 extension RefV: Equatable {
     public static func == (left: RefV, right: RefV) -> Bool {
-        return left.value == right.value
+        return left.value.id == right.value.id &&
```

Comment:
The proper implementation would be create an extension of Equatable for RefID and then do just `left.value == right.value` here.